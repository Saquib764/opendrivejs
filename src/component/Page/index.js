import './index.scss'

export default ({children, className=''})=>{
    return (
        <div data-testid='page' className={`motional-page ${className}`}>
            {children}
        </div>
    )
}