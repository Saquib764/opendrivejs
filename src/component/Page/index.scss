div.motional-page {
    text-align: left;
    display: flex;
    flex-direction: column;
    padding-left: 10px;
    padding-right: 10px;
}
