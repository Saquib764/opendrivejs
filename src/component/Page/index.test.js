import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer'
import Page from './index';

afterEach(cleanup)
test('<Page />: renders', () => {
  render(<Page />);
});

test('<Page />: renders correctly', ()=>{
  const {getByTestId} = render(<Page>My page</Page>)
  expect(getByTestId('page')).toHaveTextContent('My page')
})

test('<Page />: has class "motional-page"', ()=>{
  const {getByTestId} = render(<Page>My page</Page>)
  expect(getByTestId('page')).toHaveClass('motional-page')
})

test('<Page />: has class a custom class', ()=>{
  const {getByTestId} = render(<Page className='dashboard'>My page</Page>)
  expect(getByTestId('page')).toHaveClass('dashboard')
})

test('<Page />: has class a custom class and "motional-page"', ()=>{
  const {getByTestId} = render(<Page className='dashboard'>My page</Page>)
  expect(getByTestId('page')).toHaveClass('dashboard')
  expect(getByTestId('page')).toHaveClass('motional-page')
})

test('<Page />: test simple snapshot', ()=>{
  const tree = renderer.create(<Page className='dashboard'><div>Hi</div></Page>).toJSON()
  expect(tree).toMatchSnapshot()

})
