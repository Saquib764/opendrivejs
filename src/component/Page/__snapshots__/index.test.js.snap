// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page />: test simple snapshot 1`] = `
<div
  className="motional-page dashboard"
  data-testid="page"
>
  <div>
    Hi
  </div>
</div>
`;
